core/storage/improvements/gae_models.py

```diff
@@ -0,0 +1,194 @@
+    ENTITY_TYPE_EXPLORATION: {
+        TARGET_TYPE_STATE,
+    }
+}
+
+# Constant used to generate new IDs.
+_GENERATE_NEW_TASK_ID_MAX_ATTEMPTS = 10
+
+# Delimiter used to separate the components of a task's ID.
+_TASK_ID_PIECE_DELIMITER = '.'
+
+
+class TaskEntryModel(base_models.BaseModel):
+    """Task entry corresponding to an actionable task in the improvements tab.
+
+    Instances of a class have an ID with the form:
+        [ENTITY_TYPE].[ENTITY_ID].[TASK_TYPE].[UUID]
+    """
+    # The type of entity a task entry refers to.
+    entity_type = ndb.StringProperty(
+        required=True, indexed=True, choices=ENTITY_TYPES)
+    # The ID of the entity a task entry refers to.
+    entity_id = ndb.StringProperty(
+        required=True, indexed=True)
+    # The type of task a task entry tracks.
+    task_type = ndb.StringProperty(
+        required=True, indexed=True, choices=TASK_TYPES)
+
+    # The type of sub-entity a task entry focuses on.
+    target_type = ndb.StringProperty(
+        default=None, required=False, indexed=True, choices=TARGET_TYPES)
+    # Uniquely identifies the sub-entity a task entry focuses on.
+    target_id = ndb.StringProperty(
+        default=None, required=False, indexed=True)
+
+    # Tracks the state/progress of a task entry.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # Refers to the first entity version (inclusive) a task entry is relevant
+    # to.
+    entity_version_start = ndb.IntegerProperty(
+        required=True, indexed=True)
+    # Refers to the last entity version (exclusive) a task entry is relevant to.
+    entity_version_end = ndb.IntegerProperty(
+        default=None, required=False, indexed=True)
+    # ID of the user who closed the task, if any.
+    closed_by = ndb.StringProperty(
+        default=None, required=False, indexed=True)
+    # The date and time at which a task was closed or deprecated.
+    closed_on = ndb.DateTimeProperty(
+        default=None, required=False, indexed=True)
+    # Auto-generated string which provides a one-line summary of the task.
+    task_summary = ndb.StringProperty(
+        default=None, required=False, indexed=False)
+
+    @staticmethod
+    def get_user_id_migration_policy():
+        """TaskEntryModel has the closed_by field which refers to a user ID."""
+        return base_models.USER_ID_MIGRATION_POLICY.ONE_FIELD
+
+    @classmethod
+    def get_user_id_migration_field(cls):
+        """Return field that contains user ID."""
+        return cls.closed_by
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether any TaskEntryModel references the given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return cls.query(
+            cls.get_user_id_migration_field() == user_id).iter().has_next()
```

Comment:
Please use the field here directly since `get_user_id_migration_field` will be removed after the migration is done. Also, use `.get() is no None` instead.